bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeHandler.java

```diff
@@ -0,0 +1,388 @@
+                            jse.getMessage());
+                }
+            } else {
+                thingHandler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/offline.communication-error.empty-response");
+            }
+            if (!thingHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                thingHandler.updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            LOGGER.error("exception when receiving status packet {}", Arrays.toString(e.getStackTrace()));
+            // as we haven't received a packet we also don't know where it should have come from
+            // hence, we don't know which thing put offline.
+            // a way to monitor this would be to keep track in a list, which device answers we expect
+            // and supervise an expected answer within a given time but that will make the whole
+            // mechanism much more complicated and may be added in the future
+        } finally {
+            refreshJobRunning = false;
+        }
+    };
+
+    /**
+     * This thing related job <i>thingRefreshSender</i> triggers an update to the Govee device.
+     * The device sends it back to the common port and the response is
+     * then received by the common #refreshStatusReceiver
+     */
+    private final Runnable thingRefreshSender = () -> {
+        try {
+            triggerDeviceStatusRefresh();
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error.could-not-query-device [\"" + goveeConfiguration.hostname
+                            + "\"]");
+        }
+    };
+
+    public GoveeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        goveeConfiguration = getConfigAs(GoveeConfiguration.class);
+
+        final String ipAddress = goveeConfiguration.hostname;
+        if (!ipAddress.isEmpty()) {
+            THING_HANDLERS.put(ipAddress, this);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.ip-address.missing");
+            return;
+        }
+        if (!THING_HANDLERS.isEmpty()) {
+            startRefreshStatusJob();
+        }
+
+        if (triggerStatusJob == null) {
+            LOGGER.debug("Starting refresh trigger job for thing {} ", thing.getLabel());
+
+            triggerStatusJob = scheduler.scheduleWithFixedDelay(thingRefreshSender, 100,
+                    goveeConfiguration.refreshInterval * 1000L, TimeUnit.MILLISECONDS);
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
It would be better to set this just before scheduling the job above for 100% certainty that it won't overwrite any status set by the job.